src/android/OpenIDAuth.java

```diff
@@ -0,0 +1,218 @@
+ * for testing + to provide another exemplar of logging in properly :)
+ */
+
+class OpenIDAuth implements AuthTokenCreator {
+    private CordovaPlugin mPlugin;
+    private AuthPendingResult mAuthPending;
+    private AuthorizationService mAuthService;
+    private OpenIDAuthStateManager mStateManager;
+    private Context mCtxt;
+
+    private static final String TAG = "OpenIDAuth";
+    private static final int RC_AUTH = 100;
+
+    // This has to be a class instance instead of a singleton like in
+    // iOS because we are not supposed to store contexts in static variables
+    // singleton pattern has static GoogleAccountManagerAuth -> mCtxt
+    OpenIDAuth(Context ctxt) {
+        mCtxt = ctxt;
+        mAuthService = new AuthorizationService(mCtxt);
+        mStateManager = OpenIDAuthStateManager.getInstance(mCtxt);
+    }
+
+    @Override
+    public AuthPendingResult uiSignIn(CordovaPlugin plugin) {
+        this.mAuthPending = new AuthPendingResult();
+        this.mPlugin = plugin;
+
+        Uri issuerUri = Uri.parse(ConnectionSettings.getAuthValue(mCtxt, "discoveryURI"));
+        AuthorizationServiceConfiguration.fetchFromIssuer(
+                issuerUri,
+                new AuthorizationServiceConfiguration.RetrieveConfigurationCallback() {
+                    public void onFetchConfigurationCompleted(
+                            @Nullable AuthorizationServiceConfiguration serviceConfiguration,
+                            @Nullable AuthorizationException ex) {
+                        if (ex != null) {
+                            Log.exception(mCtxt, TAG, ex);
+                            return;
+                        }
+
+                        // use serviceConfiguration as needed
+                        // service configuration retrieved, proceed to authorization...
+                        AuthorizationRequest.Builder authRequestBuilder = new AuthorizationRequest.Builder(
+                                serviceConfiguration,
+                                ConnectionSettings.getAuthValue(mCtxt, "clientID"),
+                                ResponseTypeValues.CODE,
+                                Uri.parse("emission.auth://oauth2redirect"))
+                                .setScope(ConnectionSettings.getAuthValue(mCtxt, "scope"));
+                        AuthorizationRequest authRequest = authRequestBuilder.build();
+
+                        // AuthorizationService authService = new AuthorizationService(mCtxt);
+                        Intent authIntent = mAuthService.getAuthorizationRequestIntent(authRequest);
+                        mPlugin.cordova.setActivityResultCallback(mPlugin);
+                        mPlugin.cordova.getActivity().startActivityForResult(authIntent, RC_AUTH);
+                    }
+                });
+
+        return mAuthPending;
+    }
+
+    @Override
+    public AuthPendingResult getUserEmail() {
+        return readStoredUserEmail(mCtxt);
```

Comment:
This will return a result with (email, email) (see implementation of `readStoredUserEmail`). But the result is actually supposed to be (email, token) (see implementation in `GoogleAccountManagerAuth`). It is just for the dummy dev implementation, the token *is* the email. I wanted it to be super simple :)

I think that I actually return `null` for the token in `GoogleAccountManagerAuth.getUserEmail` so that I don't have to make a remote call just to get the email. You can do the same as well here.